infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationClientConfigListener.java

```diff
@@ -0,0 +1,189 @@
+
+    /**
+     * Tables that this stream listener will be listening to.
+     */
+    private static final List<String> tablesOfInterest = new ArrayList<>(
+            Arrays.asList(LR_REGISTRATION_TABLE_NAME, LR_MODEL_METADATA_TABLE_NAME));
+
+    /**
+     * Accessing SessionManager for client register and unregister, which could lead to session creation and termination.
+     */
+    private final SessionManager sessionManager;
+
+    /**
+     * Accessing LogReplicationConfigManager to react accordingly when new entries found in client config tables
+     */
+    private final LogReplicationConfigManager configManager;
+
+    /**
+     * CorfuStore for subscribing this StreamListener
+     */
+    private final CorfuStore corfuStore;
+
+    public LogReplicationClientConfigListener(SessionManager sessionManager, LogReplicationConfigManager configManager, CorfuStore corfuStore) {
+        super(corfuStore, CORFU_SYSTEM_NAMESPACE, CLIENT_CONFIG_TAG, tablesOfInterest);
+        this.sessionManager = sessionManager;
+        this.configManager = configManager;
+        this.corfuStore = corfuStore;
+    }
+
+    /**
+     * Subscribe this stream listener to start monitoring the changes of LR client config tables.
+     */
+    public void start() {
+        CorfuStoreMetadata.Timestamp timestamp = configManager.preprocessAndGetTail();
+        configManager.generateConfig(sessionManager.getSessions());
+
+        log.info("Start log replication listener for client config tables from {}", timestamp);
+        try {
+            corfuStore.subscribeListener(this, CORFU_SYSTEM_NAMESPACE, CLIENT_CONFIG_TAG, tablesOfInterest, timestamp);
+        } catch (StreamingException e) {
+            if (e.getExceptionCause().equals(StreamingException.ExceptionCause.LISTENER_SUBSCRIBED)) {
+                log.error("Stream listener already registered!");
+            }
+        }
+    }
+
+    /**
+     * A corfu update can/may have multiple updates belonging to different streams.
+     * This callback will return those updates as a list grouped by their Stream UUIDs.
+     * <p>
+     * Note: there is no order guarantee within the transaction boundaries.
+     *
+     * @param results is a map of stream UUID -> list of entries of this stream.
+     */
+    @Override
+    public void onNext(CorfuStreamEntries results) {
+        log.info("Client config listener onNext: {}", results.getEntries().size());
+        results.getEntries().forEach((key, value) -> {
+            log.info("Key table name: {}", key.getTableName());
+            String tableName = key.getTableName();
+            switch (tableName) {
+                case LR_REGISTRATION_TABLE_NAME:
+                    handleRegistrationTableEntries(value);
+                    break;
+                case LR_MODEL_METADATA_TABLE_NAME:
+                    handleClientMetadataTableEntries(value);
+                    break;
+                default:
+                    break;
```

Comment:
unexpected table name.  Please thrown an exception and log an error with the table name here.

Reply:
hi Pankti, I was also thinking of putting the same comment but then I realized that the only way to code coverage this path would be to mock the streaming layer because in the current implementation this callback will never receive tables it did not express interest in.

Reply:
See the point but I still have 2 questions here:
* In which case could an expected table name appear here? I think if a table that is not even in the tables of interest appears here, didn't it imply much more fundamental issue in streamingTask?
* If we throw an exception in that case, which layer/component is supposed to catch and handle it?

Reply:
yes - unless there is a fundamental issue in the StreamingTask this layer should not see an exception.
However in maybe a subsequent PR we should consider consolidating these 2 tables since we reduce the metadata overhead for the registry table since these are not high traffic tables.